parsec/backend/utils.py

```diff
@@ -13,21 +13,36 @@
 from parsec.api.version import API_V1_VERSION, API_V2_VERSION
 
 
+PEER_EVENT_MAX_WAIT = 3  # 5mn
 ALLOWED_API_VERSIONS = {API_V1_VERSION.version, API_V2_VERSION.version}
 
 
 def api(
     cmd: str,
     *,
+    long_request: bool = False,
     handshake_types: List[Union[HandshakeType, APIV1_HandshakeType]] = (
         HandshakeType.AUTHENTICATED,
         APIV1_HandshakeType.AUTHENTICATED,
     ),
 ):
     def wrapper(fn):
-        assert not hasattr(fn, "_api_info")
-        fn._api_info = {"cmd": cmd, "handshake_types": handshake_types}
-        return fn
+        if long_request:
+
+            @wraps(fn)
+            async def wrapped(self, client_ctx, *args, **kwargs):
+                with trio.move_on_after(PEER_EVENT_MAX_WAIT):
```

Comment:
I don't think we should have this timeout here.

This is part of the apiv1invite/claim commands, but hasn't been added to the apiv2 commands (from what I saw in the code there is absolutely no timeout handilng in the backend, the core or the gui regarding apiv2 invite/claim process).

I think if a timeout has to be implemented, we'd better have it on the client side, typically in the GUI (given it has to deal anyway with user getting bored and manually cancelling the invite/claim by himself)

On top of that `events_listen` is another long time command, but should never timeout given we know receiving events is an infrequent operation, otherwise we would spend most of the socket traffic retrying command after a timeout... (note backend keep a buffer of to-send-to-peer events, so that timeout request won't make the client miss an event arriving to the backend at the wrong).

Reply:
@vxgmichel any idea about this ?